### PR TITLE
Add a roadmap

### PR DESCRIPTION
**Pull request by j-f1**
This adds a roadmap with a complete listing of TSLint core rules and their equivalents in the ESLint ecosystem.

[**Rendered Version 📄**](https://github.com/bradzacher/eslint-plugin-typescript/blob/add-roadmap/ROADMAP.md)

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+| [`object-literal-shorthand`]        |     | [`typescript/object-literal-shorthand`]()        |
+| [`one-line`]                        |     | [`typescript/one-line`]()                        |
+| [`one-variable-per-declaration`]    |     | [`typescript/one-variable-per-declaration`]()    |
+| [`ordered-imports`]                 |     | [`typescript/ordered-imports`]()                 |
+| [`prefer-function-over-method`]     |     | [`typescript/prefer-function-over-method`]()     |
+| [`prefer-method-signature`]         |     | [`typescript/prefer-method-signature`]()         |
+| [`prefer-switch`]                   |     | [`typescript/prefer-switch`]()                   |
+| [`prefer-template`]                 |     | [`typescript/prefer-template`]()                 |
+| [`prefer-while`]                    |     | [`typescript/prefer-while`]()                    |
+| [`quotemark`]                       |     | [`typescript/quotemark`]()                       |
```
Comment (armano2) [inline ROADMAP.md]: https://eslint.org/docs/rules/quotes

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+| TSLint rule                         |     | ESLint rule                                      |
+| ----------------------------------- | :-: | ------------------------------------------------ |
+| [`cyclomatic-complexity`]           |     | [`typescript/cyclomatic-complexity`]()           |
+| [`deprecation`]                     |     | [`typescript/deprecation`]()                     |
+| [`eofline`]                         |     | [`typescript/eofline`]()                         |
+| [`indent`]                          |     | [`typescript/indent`]()                          |
+| [`linebreak-style`]                 |     | [`typescript/linebreak-style`]()                 |
+| [`max-classes-per-file`]            |     | [`typescript/max-classes-per-file`]()            |
+| [`max-file-line-count`]             |     | [`typescript/max-file-line-count`]()             |
```
Comment (armano2) [inline ROADMAP.md]: https://eslint.org/docs/rules/max-lines

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+| [`prefer-switch`]                   |     | [`typescript/prefer-switch`]()                   |
+| [`prefer-template`]                 |     | [`typescript/prefer-template`]()                 |
+| [`prefer-while`]                    |     | [`typescript/prefer-while`]()                    |
+| [`quotemark`]                       |     | [`typescript/quotemark`]()                       |
+| [`return-undefined`]                |     | [`typescript/return-undefined`]()                |
+| [`semicolon`]                       |     | [`typescript/semicolon`]()                       |
+| [`space-before-function-paren`]     |     | [`typescript/space-before-function-paren`]()     |
+| [`space-within-parens`]             |     | [`typescript/space-within-parens`]()             |
+| [`switch-final-break`]              |     | [`typescript/switch-final-break`]()              |
+| [`type-literal-delimiter`]          |     | [`typescript/type-literal-delimiter`]()          |
```
Comment (armano2) [inline ROADMAP.md]: typescript/member-delimiter-style

File: ROADMAP.md [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+| [`one-variable-per-declaration`]    |     | [`typescript/one-variable-per-declaration`]()    |
+| [`ordered-imports`]                 |     | [`typescript/ordered-imports`]()                 |
+| [`prefer-function-over-method`]     |     | [`typescript/prefer-function-over-method`]()     |
+| [`prefer-method-signature`]         |     | [`typescript/prefer-method-signature`]()         |
+| [`prefer-switch`]                   |     | [`typescript/prefer-switch`]()                   |
+| [`prefer-template`]                 |     | [`typescript/prefer-template`]()                 |
+| [`prefer-while`]                    |     | [`typescript/prefer-while`]()                    |
+| [`quotemark`]                       |     | [`typescript/quotemark`]()                       |
+| [`return-undefined`]                |     | [`typescript/return-undefined`]()                |
+| [`semicolon`]                       |     | [`typescript/semicolon`]()                       |
```
Comment (armano2) [inline ROADMAP.md]: https://github.com/bradzacher/eslint-plugin-typescript/blob/master/docs/rules/member-delimiter-style.md

File: ROADMAP.md [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+| [`number-literal-format`]           |     | [`typescript/number-literal-format`]()           |
+| [`object-literal-key-quotes`]       |     | [`typescript/object-literal-key-quotes`]()       |
+| [`object-literal-shorthand`]        |     | [`typescript/object-literal-shorthand`]()        |
+| [`one-line`]                        |     | [`typescript/one-line`]()                        |
+| [`one-variable-per-declaration`]    |     | [`typescript/one-variable-per-declaration`]()    |
+| [`ordered-imports`]                 |     | [`typescript/ordered-imports`]()                 |
+| [`prefer-function-over-method`]     |     | [`typescript/prefer-function-over-method`]()     |
+| [`prefer-method-signature`]         |     | [`typescript/prefer-method-signature`]()         |
+| [`prefer-switch`]                   |     | [`typescript/prefer-switch`]()                   |
+| [`prefer-template`]                 |     | [`typescript/prefer-template`]()                 |
```
Comment (armano2) [inline ROADMAP.md]: https://eslint.org/docs/rules/prefer-template

File: ROADMAP.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+| [`newline-per-chained-call`]        |     | [`typescript/newline-per-chained-call`]()        |
+| [`new-parens`]                      |     | [`typescript/new-parens`]()                      |
+| [`no-angle-bracket-type-assertion`] |     | [`typescript/no-angle-bracket-type-assertion`]() |
+| [`no-boolean-literal-compare`]      |     | [`typescript/no-boolean-literal-compare`]()      |
+| [`no-consecutive-blank-lines`]      |     | [`typescript/no-consecutive-blank-lines`]()      |
+| [`no-irregular-whitespace`]         |     | [`typescript/no-irregular-whitespace`]()         |
+| [`no-parameter-properties`]         |     | [`typescript/no-parameter-properties`]()         |
+| [`no-redundant-jsdoc`]              |     | [`typescript/no-redundant-jsdoc`]()              |
+| [`no-reference-import`]             |     | [`typescript/no-reference-import`]()             |
+| [`no-trailing-whitespace`]          |     | [`typescript/no-trailing-whitespace`]()          |
```
Comment (armano2) [inline ROADMAP.md]: https://eslint.org/docs/rules/no-trailing-spaces

**Comment by armano2** [comment]
@j-f1 `eslint-plugin-import` is not supporting typescript equal assign import and export assign import
```
import foo = require('bar') 
```

**Comment by j-f1** [comment]
https://github.com/benmosher/eslint-plugin-import/issues/1244 @armano2

**Comment by armano2** [comment]
@bradzacher is currently working on indent rule see https://github.com/bradzacher/eslint-plugin-typescript/pull/219

**Comment by armano2** [comment]
`file-header`:
- https://github.com/Stuk/eslint-plugin-header
- https://github.com/Sekhmet/eslint-plugin-file-header

**Comment by armano2** [comment]
i'm going to work on `interface-over-type-literal`

**Comment by bradzacher** [comment]
That being said, it's great for us to cross promote other packages.
The more work we can reasonably offload to existing packages the better. Love the cross promotion of packages like import.

That being said, while I do love prettier [1], but due to its low configurability I think we should be cautious about not implementing things here purely because it's in prettier. Though alerting people to it as an alternative to fill our holes is a great idea.

---

[1] since starting at FB (where it's on by default for JS code), I've since added prettier-eslint to all my personal projects. For my code I didn't quite like how prettier formatted the code, but when paired with eslint fixers it's perfect. It's literally changed my coding life.

**Pull request merged**